### DISCUSSION TITLE
How would I pipe commands with no hardcoding of command strings?

### DISCUSSION THREAD

**Discussion by redragonx** [rust-shell-script/rust_cmd_lib#33] [General]
Hi, I would like to know how you can combine commands without hardcoding the actual cmd in the macro?

Not sure how escaping works in these cases.

Rough Example:
```rust
        let post_cmd  = "grep 'blah' | another_cmd"
        let output_raw = self.someVarEval();

        let data = run_fun!(echo "$output_raw" | $post_cmd );
```

When I run the above in full code. I get the following output error:

```
No such file or directory (os error 2)

**Comment by rust-shell-script** [comment]
You can construct command string in your way and pass it to bash:
```
let cmd_str = ....;
let data = run_fun!(bash -c $cmd_str)?;
```

It is not encouraged to do things like this, since it could introduce command injection and also you need to make the correct quotes. But anyway you are already doing things like that, so take your own risk.

**Reply by redragonx** [reply]
Thanks. I figured as much that I had to run bash -c

Badass library btw.
